src/main/java/org/prebid/server/bidder/ix/IxBidder.java

```diff
@@ -290,8 +292,32 @@ private static Response mergeNativeImpTrackers(Response response, List<EventTrac
                 .build();
     }
 
-    private static BidType getBidType(String impId, List<Imp> imps) {
-        for (Imp imp : imps) {
+    private static BidType getBidType(Bid bid, List<Imp> imps) {
+        final BidType bidType = getBidTypeFromMtype(bid.getMtype());
+        if (bidType != null) {
+            return bidType;
+        }
+
+        return Optional.ofNullable(bid.getExt())
+                .map(ext -> ext.get("prebid"))
+                .map(prebid -> prebid.get("type"))
+                .map(JsonNode::asText)
+                .map(BidType::fromString)
+                .orElse(getBidTypeFromImp(imps, bid.getImpid()));
```

Comment:
If you are using `orElse` argument value will always be calculated, to had it calculated only if really needed
pls use (.orElseGet(() -> getBidTypeFromImp(imps, bid.getImpid())))